test/realm_set_test.dart

```diff
@@ -0,0 +1,535 @@
+        return Sets(intSet as RealmSet<int>, [-1, 0, 1]);
+      case String:
+        return Sets(stringSet as RealmSet<String>, ['Tesla', 'VW', 'Audi']);
+      case double:
+        return Sets(doubleSet as RealmSet<double>, [-1.1, 0.1, 1.1, 2.2, 3.3, 3.14]);
+      case DateTime:
+        return Sets(dateTimeSet as RealmSet<DateTime>, [DateTime(2023).toUtc(), DateTime(1981).toUtc()]);
+      case ObjectId:
+        return Sets(objectIdSet as RealmSet<ObjectId>, [ObjectId.fromTimestamp(DateTime(2023).toUtc()), ObjectId.fromTimestamp(DateTime(1981).toUtc())]);
+      case Uuid:
+        return Sets(uuidSet as RealmSet<Uuid>, [Uuid.fromString("12345678123456781234567812345678"), Uuid.fromString("82345678123456781234567812345678")]);
+      case RealmValue:
+        return Sets(mixedSet as RealmSet<RealmValue>, [RealmValue.nullValue(), RealmValue.bool(true), RealmValue.int(1), RealmValue.string("text")]);
+      case RealmObject:
+        return Sets(objectsSet as RealmSet<Car>, [Car("Tesla"), Car("VW"), Car("Audi")], (realm, value) => realm.find<Car>((value as Car).make));
+      case _NullableBool:
+        return Sets(nullableBoolSet as RealmSet<bool?>, [...setByType(bool).values, null]);
+      case _NullableInt:
+        return Sets(nullableIntSet as RealmSet<int?>, [...setByType(int).values, null]);
+      case _NullableString:
+        return Sets(nullableStringSet as RealmSet<String?>, [...setByType(String).values, null]);
+      case _NullableDouble:
+        return Sets(nullableDoubleSet as RealmSet<double?>, [...setByType(double).values, null]);
+      case _NullableDateTime:
+        return Sets(nullableDateTimeSet as RealmSet<DateTime?>, [...setByType(DateTime).values, null]);
+      case _NullableObjectId:
+        return Sets(nullableObjectIdSet as RealmSet<ObjectId?>, [...setByType(ObjectId).values, null]);
+      case _NullableUuid:
+        return Sets(nullableUuidSet as RealmSet<Uuid?>, [...setByType(Uuid).values, null]);
+      default:
+        throw RealmError("Unsupported type $type");
+    }
+  }
+
+  List<Object?> values(Type type) {
+    return setByType(type).values;
+  }
+
+  List<Object?> getValuesOrManagedValues(Realm realm, Type type) {
+    Sets set = setByType(type);
+    if (set.values.first is! RealmObject) {
+      return set.values;
+    }
+
+    return set.values.map<Object?>((value) {
+      RealmObject? realmValue = set.getRealmObject!(realm, value as RealmObject);
+      return realmValue;
+    }).toList();
+  }
+}
+
+class Sets {
+  final RealmSet<Object?> set;
+  final List<Object?> values;
+  RealmObject? Function(Realm realm, RealmObject value)? getRealmObject = (realm, value) => value;
+
+  Sets(this.set, this.values, [this.getRealmObject]);
+}
+
+Future<void> main([List<String>? args]) async {
+  await setupTests(args);
+
+  for (var type in supportedTypes) {
+    test('RealmSet<$type> unmanged set add', () {
+      final testSet = TestRealmSets(1);
+      final set = testSet.setByType(type).set;
```

Comment:
The values are initialized on this row, but they are not used. They are initialized on the next row again.